stock_by_warehouse/static/src/js/widget.js

```diff
@@ -2,45 +2,96 @@ odoo.define('stock_by_warehouse.warehouse', function (require) {
 "use strict";
 
 var core = require('web.core');
+var Widget = require('web.Widget');
 var form_common = require('web.AbstractField');
 var formats = require('web.field_utils');
 var fieldRegistry = require('web.field_registry');
 var QWeb = core.qweb;
 
+var Locations = Widget.extend({
+    init: function (info) {
+        this.info = info;
+    },
 
-var ShowPaymentLineWidget = form_common.extend({
-    _render: function() {
-        this._super.apply(this, arguments);
-        var info = JSON.parse(this.value);
-        this.$el.html(QWeb.render('ShowWarehouseInfo', {
-            'show':  formats.format["float"](info.warehouse || "", this.field, {digits: info.digits || ""})
-        }));
-        var button = this.$('.js_product_warehouse');
+    start: function () {
+        var info = this.info;
         if (info !== false) {
             _.each(info.content, function (k, v) {
                 k.index = v;
-                k.available_not_res_formated = formats.format["float"](k.available_not_res, "", {digits: k.digits});
-                k.available_formated  = formats.format["float"](k.available, "", {digits: k.digits});
-                k.incoming_formated  = formats.format["float"](k.incoming, "", {digits: k.digits});
-                k.outgoing_formated  = formats.format["float"](k.outgoing, "", {digits: k.digits});
-                k.virtual_formated  = formats.format["float"](k.virtual, "", {digits: k.digits});
-                k.saleable_formated  = formats.format["float"](k.saleable, "", {digits: k.digits});
+                k.name = k.warehouse_name
+                k.locations_quantity_formatted = formats.format["integer"](
+                    k.locations_available, "", {digits: k.digits});
+                k.locations_info = k.info_content;
             });
         }
-        var popover = QWeb.render('ProductWarehousePopOver', {
-            'lines': info.content || []
+        var popover = QWeb.render('StockAvailabilityPopOver', {
+            'lines': info.content || [],
         });
+
         var options = {
-            'content': popover,
-            'html': true,
-            'placement': this.nodeOptions.placement || 'right',
-            'title': this.nodeOptions.title || info.title || "",
-            'trigger': 'focus',
-            'delay': { "show": 0, "hide": 100 }
+            content: popover,
+            html: true,
+            placement: 'right',
+            title: info.title || "",
+            trigger: 'focus',
+            delay: {"show": 0, "hide": 100},
         };
-        $(button).popover(options);
+        this.$el.popover(options);
+    },
+});
+
+var ShowPaymentLineWidget = form_common.extend({
+    _render: function() {
+        var self = this;
+        this._super.apply(this, arguments);
+        this.info = JSON.parse(this.value);
+
+        if (this.nodeOptions.by_location){
+            this.$el.html(QWeb.render('StockAvailabilityAlert', {
```

Comment:
I think qweb template names should be prefixed with the module name

Reply:
js qweb templates do not manage namespaces (that for what the prefix is used for) that's only on the backend ones.

Reply:
Resolving since it's not as used.